formatter/src/formatter/expr.rs

```diff
@@ -1,35 +1,59 @@
-use syn::ExprLit;
-use syn_rsx::NodeValueExpr;
+use syn::{Block, Expr, ExprBlock, ExprLit, LitStr};
 
 use crate::{formatter::Formatter, view_macro::ViewMacroFormatter};
 
 impl Formatter<'_> {
+    pub fn literal_str(&mut self, string: &LitStr) {
+        self.expr(&Expr::Lit(ExprLit {
```

Comment:
speaking of, if you use RawText or Text, what is the multiline behaviour? Does it include the indentation, I guess so right?

The reason I ask is because of #33 and #7

I feel like we have to implement something similar to the behaviour of `indoc!` to strip away the indentation.

Reply:
@bram209 `RawText` is just a `TokenStream`with source_text information (on nightly).

So it saves any spaces/newlines/tabs and other invisible symbols inside raw text (and optionally including whitespaces).

I think it can be used to fix #33, but probably we need to embed fmt for js/css or implement generic "indentation based formatting"

> I think in this case it is fine to do self.printer.word(string.value());

This one don't include quotes ("") so i used code from your second comment.

Reply:
I think we should do something similar to `indoc!` and un-ident the text so that the leftmost non-space character is in the first column, while perserving the indents relative to that leftmost character.

Reply:
> will be "    Hello\n    World" when including whitespace?
true

> And "Hello\nWorld" excluding whitespace?

No, it will be `"Hello\n    World"` (single RawText node). Identation inside `RawText` treated as part of string.

But if you have:
```
<div>
    Hello World
    "Foo"
</div>
```
including whitespaces RawText = `"    Hello World\n    "`.
And excluding them, it will be RawText = `"Hello World"`.

Also, there some places, where it is impossible to recover whitespaces, for example, when RawText is used outside of other elements, therefore:
```
      Hello
      World
```

including whitespaces will return `None`,
and excluding will return `Some("Hello\n      World")`

Internally it tries to recover whitespaces by getting tokens before and after raw text, and getting source_text of all that span.

>I think we should do something similar to indoc! and un-ident the text so that the leftmost non-space character is in the first column, while perserving the indents relative to that leftmost character.

I will check it, maybe it would be a good idea to embed this into RawText method.